crates/ibc/src/core/ics04_channel/handler/recv_packet.rs

```diff
@@ -128,12 +127,21 @@ pub(crate) mod val_exec_ctx {
                 validate_write_acknowledgement(ctx_b, msg)?;
             }
         } else {
-            ctx_b.get_packet_receipt(&(
+            let packet_rec = ctx_b.get_packet_receipt(&(
                 msg.packet.port_on_b.clone(),
                 msg.packet.chan_on_b.clone(),
                 msg.packet.sequence,
-            ))?;
-
+            ));
+            match packet_rec {
+                Ok(_receipt) => {}
+                Err(ContextError::PacketError(PacketError::PacketReceiptNotFound { sequence }))
+                    if sequence == msg.packet.sequence => {}
+                Err(_) => {
+                    return Err(ContextError::PacketError(
+                        PacketError::ImplementationSpecific,
+                    ))
```

Comment:
why discard the original error here?

Reply:
to comply with [here](https://github.com/cosmos/ibc-rs/blob/HEAD/crates/ibc/src/core/ics04_channel/handler/recv_packet.rs#L323). Maybe I should fix this line instead?

Reply:
yes; I don't like the "implementation specific" error

Reply:
Agree, done!